gcc 10.x compile warning/error for array subscript is outside the bounds in cmsis_rtos_v2/thread.c

When building `samples/portability/cmsis_rtos_v2/philosophers/sample.portability.cmsis_rtos_v2.philosopher` via sanitycheck with gcc 10.2 we get:

```
Scanning dependencies of target lib__cmsis_rtos_v2
[ 84%] Building C object zephyr/lib/cmsis_rtos_v2/CMakeFiles/lib__cmsis_rtos_v2.dir/thread.c.obj
/home/galak/git/zephyr/lib/cmsis_rtos_v2/thread.c: In function 'osThreadNew':
/home/galak/git/zephyr/lib/cmsis_rtos_v2/thread.c:172:11: error: array subscript this_thread_num_dynamic is outside array bounds of 'struct z_thread_stack_element[0][512]' [-Werror=array-bounds]
  172 |   stack = cv2_thread_stack_pool[this_thread_num_dynamic];
      |           ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/galak/git/zephyr/include/kernel_includes.h:44,
                 from /home/galak/git/zephyr/include/kernel.h:17,
                 from /home/galak/git/zephyr/lib/cmsis_rtos_v2/thread.c:7:
/home/galak/git/zephyr/lib/cmsis_rtos_v2/thread.c:32:36: note: while referencing 'cv2_thread_stack_pool'
   32 | static K_THREAD_STACK_ARRAY_DEFINE(cv2_thread_stack_pool,       \
      |                                    ^~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/sys/thread_stack.h:148:3: note: in definition of macro 'K_KERNEL_STACK_ARRAY_DEFINE'
  148 |   sym[nmemb][Z_KERNEL_STACK_LEN(size)]
      |   ^~~
/home/galak/git/zephyr/lib/cmsis_rtos_v2/thread.c:32:8: note: in expansion of macro 'K_THREAD_STACK_ARRAY_DEFINE'
   32 | static K_THREAD_STACK_ARRAY_DEFINE(cv2_thread_stack_pool,       \
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

--- a/lib/cmsis_rtos_v2/thread.c
+++ b/lib/cmsis_rtos_v2/thread.c
@@ -29,9 +29,11 @@ static struct cv2_thread cv2_thread_pool[CONFIG_CMSIS_V2_THREAD_MAX_COUNT];
 static uint32_t thread_num;
 static uint32_t thread_num_dynamic;
 
+#if CONFIG_CMSIS_V2_THREAD_DYNAMIC_MAX_COUNT != 0
 static K_THREAD_STACK_ARRAY_DEFINE(cv2_thread_stack_pool,		     \
 				   CONFIG_CMSIS_V2_THREAD_DYNAMIC_MAX_COUNT, \
 				   CONFIG_CMSIS_V2_THREAD_DYNAMIC_STACK_SIZE);
+#endif
 
 static inline int _is_thread_cmsis_inactive(struct k_thread *thread)
 {
@@ -108,7 +110,6 @@ osThreadId_t osThreadNew(osThreadFunc_t threadfunc, void *arg,
 	void *stack;
 	size_t stack_size;
 	uint32_t this_thread_num;
-	uint32_t this_thread_num_dynamic;
 
 	if (k_is_in_isr()) {
 		return NULL;
@@ -163,14 +164,18 @@ osThreadId_t osThreadNew(osThreadFunc_t threadfunc, void *arg,
 	tid = &cv2_thread_pool[this_thread_num];
 	tid->attr_bits = attr->attr_bits;
 
+#if CONFIG_CMSIS_V2_THREAD_DYNAMIC_MAX_COUNT != 0
 	if (attr->stack_mem == NULL) {
+		uint32_t this_thread_num_dynamic;
 		__ASSERT(CONFIG_CMSIS_V2_THREAD_DYNAMIC_STACK_SIZE > 0,
 			 "dynamic stack size must be configured to be non-zero\n");
 		this_thread_num_dynamic =
 			atomic_inc((atomic_t *)&thread_num_dynamic);
 		stack_size = CONFIG_CMSIS_V2_THREAD_DYNAMIC_STACK_SIZE;
 		stack = cv2_thread_stack_pool[this_thread_num_dynamic];
-	} else {
+	} else
+#endif
+	{
 		stack_size = attr->stack_size;
 		stack = attr->stack_mem;
 	}

